actor/builtin/miner/miner.go

```diff
@@ -671,9 +667,9 @@ func (ma *Actor) SubmitPoSt(ctx exec.VMContext, postProofs []proofs.PoStProof) (
 			sectorStoreType = proofs.Test
 		}
 
-		seed, err := ma.currentProvingPeriodPoStChallengeSeed(ctx)
+		seed, err := currentProvingPeriodPoStChallengeSeed(ctx, state)
 		if err != nil {
-			return nil, errors.RevertErrorWrap(err, "failed to sample chain for challenge seed")
+			return nil, errors.FaultErrorWrap(err, "failed to sample chain for challenge seed")
```

Comment:
I think Fault is wrong here.  I believe this should be a [temporary error](https://github.com/filecoin-project/go-filecoin/blob/master/vm/errors/errors.go#L202) in the case that the sample height is too high, and a permanent error in other cases.  cc @acruikshank.

Reply:
@ZenGround0 (cc @acruikshank) I will file a follow-up issue for this.